### ISSUE TITLE
crash withe basic call to HELP : help, out='will_crash'

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1102]
```
(u1404)
GDL> help, out='tototototo'
*** Error in `/home/alaingdl/GDL/last-gdl/src/gdl': free(): invalid pointer: 0x00000000029ed8a0 ***

(debian 10 and U2004)
GDL> help, out='tototototo'
free(): invalid pointer
Magick: abort due to signal 6 (SIGABRT) "Abort"...
```

**Comment by alaingdl** [comment]
Same issue may appear (with crash or no message), is in : ` a=histogram(findgen(10), loc='bad')`
which should issue a warning like : 
`% HISTOGRAM: Expression must be named variable in this context: <STRING   ('bad')>.`

**Comment by GillesDuvert** [comment]
One-liner? This is important.

**Comment by alaingdl** [comment]
I am working on that, with progress. I can just fix it locally, but I can also consider more generally
the situation where Keyword should not be a so-called "named variable"

The problem is that some good old code (in fact GetString in envt.cpp) is not that simple to cleanup

 const string EnvBaseT::GetString( BaseGDL*& p, bool calledFromHELP)

**Comment by alaingdl** [comment]
And I would say #1089 is way more annoying because HISTOGRAM() is frequently used, and `test_histo.pro`
covers nothing ...

**Comment by GillesDuvert** [comment]
histogram needs to be rewritten entirely at some point as it is 20 times slower than IDL. GSL is not known for its speed, but here the difference is striking.

**Comment by GillesDuvert** [comment]
The crash was inavoidable as the programer did not check the variable was 'writeable'.
This because the functions to access variables are not well defined enough. One should insure a set of easy functions that make the difference between local and global variables as well as readonly and writeable variables (and prabably the boolean variables).
Nothing difficult, WIP.

**Comment by GillesDuvert** [comment]
closed by #1174
